File: java/src/main/java/io/fivexer/sdk/model/WorkerAvailability.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * The result of pausing or resuming a worker.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerAvailability {
    private String id;
    private boolean available;
    private List<String> releasedTaskIds;  // only present when the pause requested a backlog release

    public String getId() { return id; }
    public boolean isAvailable() { return available; }
    public List<String> getReleasedTaskIds() { return releasedTaskIds; }
}

File: php/src/Model/WorkerAvailability.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The result of pausing or resuming a worker.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerAvailability
{
    public function __construct(
        public readonly string $id,
        public readonly bool $available,
        /** @var list<string>|null */
        public readonly ?array $releasedTaskIds,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            available: (bool) ($data['available'] ?? false),
            releasedTaskIds: isset($data['releasedTaskIds']) ? \array_map('strval', $data['releasedTaskIds']) : null,
        );
    }
}
